osu.Desktop/Windows/GameplayWinKeyHandler.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Platform;
+using osu.Game.Configuration;
+
+namespace osu.Desktop.Windows
+{
+    public class GameplayWinKeyHandler : Component
+    {
+        private Bindable<bool> allowScreenSuspension;
+        private Bindable<bool> disableWinKey;
+
+        private GameHost host;
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost host, OsuConfigManager config)
+        {
+            this.host = host;
+
+            allowScreenSuspension = host.AllowScreenSuspension.GetBoundCopy();
+            allowScreenSuspension.ValueChanged += toggleWinKey;
```

Comment:
My reservations towards this whole thing aside, I find this awkward given that `AllowScreenSuspension` was originally intended for mobile platforms, so now one setting has two semantic meanings on two different platforms.

Reply:
Yeah I agree. Should just use the `GameplayDisableWinKey` setting and call it a day. Probably should be something more like `ScreenSuspensionHandler` that's added to the hierarchy whenever needed rather than being at the `Game` level.

Reply:
Hmm, are you proposing rewriting the screen suspension logic osu!-side? Or saying that the windows key isn't always going to be disabled when screen suspension is disabled (potentially editor?)

Reply:
> saying that the windows key isn't always going to be disabled when screen suspension is disabled

This basically. Editor (maybe), replays, match screen (while in a live room), probably will disable screen suspension but shouldn't disable windows key.

Reply:
Fair enough. So the plan is to dial back and disable windows key game-wide? If that's the case, the default needs to be `false` but it will need to be reset to a new setting when we change it to be player-only, and the default should be `true`. Also the setting text is wrong if we are going to do this.

I'd propose it should be fixed in this PR to only occur during gameplay.

Reply:
My initial implementation (wrongly) assumed that all the screens that disabled screen suspension would disable the windows key as well (which in practice isn't right at all, according to smoogipoo's previous answer).

I was thinking of extending `ScreenSuspensionHandler` with a bindable boolean to allow disabling the windows key independently of the screen suspension and add a way to have the state globally available through caching (basically an instance would be globally cached and the instances added to a screen drawable hierarchy would bind to the global instance bindables). The `osu.Desktop` part would just have to bind to the bindables of the globally cached instance.

Using alternatively a `SessionStatic` may work aswell.